components/Author/Profile/AuthorProfileHeader.tsx

```diff
@@ -120,24 +120,39 @@ const AuthorProfileHeader = () => {
     revalidateAuthorProfile(profile.id);
   };
 
+  
+  const [isShowingAll, setIsShowingAll] = useState(false);
+  const [shouldSeeSuspensionBanner, setShouldSeeSuspensionBanner] = useState(false);
+  const [isShowingFullDescription, setIsShowingFullDescription] = useState(false);
+  const visibleInstitutions = isShowingAll ? profile.education : profile.education.slice(0, 1);
+  const truncatedDescription = truncateText(profile.description, 300);
+  const hasEducation = profile.education.length > 0 || profile.institutions.length > 0;
+  
   useEffect(() => {
     (async () => {
       // This is necessary in order to have author data appear in the "edit profile" modal
       await dispatch(AuthorActions.getAuthor({ authorId: profile.id }));
     })();
   }, []);
 
-  const [isShowingAll, setIsShowingAll] = useState(false);
-  const [isShowingFullDescription, setIsShowingFullDescription] = useState(false);
-  const visibleInstitutions = isShowingAll ? profile.education : profile.education.slice(0, 1);
-  const truncatedDescription = truncateText(profile.description, 300);
-  const hasEducation = profile.education.length > 0 || profile.institutions.length > 0;
+  useEffect(() => {
+    if (currentUser) {
+      setShouldSeeSuspensionBanner(
+        Boolean(
+          (profile.user?.isProbableSpammer || profile.user?.isSuspended) && (currentUser?.moderator || currentUser?.editorOf)
```

Comment:
Aren't you already checking `profile.user?.isProbableSpammer || profile.user?.isSuspended` in UserStateBanner?

Reply:
Hey you know what, this is a legacy component and somehow totally overlooked this. Thanks for pointing this out.